### PR TITLE
Hide not allowed Reactions

### LINKED ISSUE #9286: Hide old Reactions wich are no longer Allowed
This issue colect options of the gitea communety about title topic

### PR DESCRIPTION
**Pull request by 6543**
close #9286

come up with #8886

### CODE REVIEW

**Comment by 6543** [comment]
If someone know how to add this as xorm condition, is welcome to tell me the solution

**Review by zeripath — APPROVED** [review]

File: models/issue_reaction.go [hunk 1]
```diff
@@ -68,9 +68,18 @@ func FindIssueReactions(issue *Issue) (ReactionList, error) {
 func findReactions(e Engine, opts FindReactionsOptions) ([]*Reaction, error) {
 	reactions := make([]*Reaction, 0, 10)
 	sess := e.Where(opts.toConds())
```
Comment (lafriks) [inline models/issue_reaction.go:70]: This should work:
```
.In("reaction.`type`", setting.UI.ReactionsMap key array)
```

Comment (6543) [reply models/issue_reaction.go:70]: @lafriks figure it out - thanks the `.In(...)` hint was it :d - was also writen in discord but had no time until jet to change it

-> DONE

**Comment by 6543** [comment]
@lafriks done

File: models/issue_reaction.go (hunk 1 above)
Comment (axifive) [reply models/issue_reaction.go:70]: @6543 Better use ``` .`type` ``` as @lafriks suggested, Postgres has `type` keyword

File: models/issue_reaction.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ func findReactions(e Engine, opts FindReactionsOptions) ([]*Reaction, error) {
...
 	return reactions, sess.
+		In("reaction.type", setting.UI.Reactions).
```
Comment (axifive) [inline models/issue_reaction.go]:
```suggestion
		In("reaction.`type`", setting.UI.Reactions).
```

Comment (6543) [reply models/issue_reaction.go]: this is pointles - if so i have to change the rest too - it works without :D just code format

Comment (6543) [reply models/issue_reaction.go]: @axifive i can add if but otherwise i resolve

Comment (axifive) [reply models/issue_reaction.go]: @6543 I commented above

Comment (6543) [reply models/issue_reaction.go]: @axifive i dont see `added value` but if you can explain why i should add "`" I'll do :D

Comment (6543) [reply models/issue_reaction.go]: @axifive I wrote you a longer text on discord - to dont spam github

Comment (6543) [reply models/issue_reaction.go]: sort i mean: [link](https://github.com/go-gitea/gitea/pull/9387/files#diff-61ce589b0f29ee34fc2448fcda3af467L72)

Comment (lafriks) [reply models/issue_reaction.go]: `type` is reserved keyword so it is safer to escape it to be on safe side on all databases and versions :)

Comment (6543) [reply models/issue_reaction.go]: Thx - missing part 👍

**Review by axifive — APPROVED** [review]

**Comment by 6543** [comment]
@lafriks @axifive now gitea is kind a broken ;( do xorm escape on its own?

**Comment by lafriks** [comment]
@6543 that is xorm bug that has been already fixed upstream. Just update xorm:
```
go get -u xorm.io/xorm@c349d36e950a265a89b1d71e16c7583100861a5c
go mod vendor
```

@lunny can you tag xorm a new version?

**Comment by lunny** [comment]
@lafriks @6543 I have tagged v0.8.1 https://gitea.com/xorm/xorm/releases

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
ready to :rocket:

**Pull request merged**